pkg/providers/tencent/tencent.go

```diff
@@ -0,0 +1,1431 @@
+		return err
+	}
+
+	p.logger.Infof("[%s] successfully executed start logic\n", p.GetProviderName())
+
+	return nil
+}
+
+func (p *Tencent) StopK3sCluster(f bool) error {
+	p.logger = common.NewLogger(common.Debug)
+	p.logger.Infof("[%s] executing stop logic...\n", p.GetProviderName())
+
+	if err := p.generateClientSDK(); err != nil {
+		return err
+	}
+
+	if err := p.stopCluster(f); err != nil {
+		return err
+	}
+
+	p.logger.Infof("[%s] successfully executed stop logic\n", p.GetProviderName())
+
+	return nil
+}
+
+func (p *Tencent) SSHK3sNode(ssh *types.SSH) error {
+	p.logger = common.NewLogger(common.Debug)
+	p.logger.Infof("[%s] executing ssh logic...\n", p.GetProviderName())
+
+	if err := p.generateClientSDK(); err != nil {
+		return err
+	}
+
+	response, err := p.describeInstances()
+	if err != nil {
+		return err
+	}
+	if len(response.Response.InstanceSet) < 1 {
+		return fmt.Errorf("[%s] calling preflight error: cluster name `%s` do not exist",
+			p.GetProviderName(), p.Name)
+	}
+
+	ids := make(map[string]string, len(response.Response.InstanceSet))
+	for _, instance := range response.Response.InstanceSet {
+		if instance.PublicIpAddresses != nil {
+			for _, t := range instance.Tags {
+				switch *t.Key {
+				case "master":
+					if *t.Value == "true" {
+						ids[*instance.InstanceId] = *instance.PublicIpAddresses[0] + " (master)"
+					}
+				case "worker":
+					if *t.Value == "true" {
+						ids[*instance.InstanceId] = *instance.PublicIpAddresses[0] + " (worker)"
+					}
+				default:
+					continue
+				}
+			}
+		}
+	}
+
+	ip := strings.Split(utils.AskForSelectItem(fmt.Sprintf("[%s] choose ssh node to connect", p.GetProviderName()), ids), " (")[0]
+
+	if ip == "" {
+		return fmt.Errorf("[%s] choose incorrect ssh node", p.GetProviderName())
+	}
+
+	c := &types.Cluster{
+		Metadata: p.Metadata,
+		Options:  p.Options,
+		Status:   p.Status,
+	}
+
+	// ssh K3s node.
+	if err := cluster.SSHK3sNode(ip, c, ssh); err != nil {
+		return err
+	}
+
+	p.logger.Infof("[%s] successfully executed ssh logic\n", p.GetProviderName())
+
+	return nil
+}
+
+func (p *Tencent) IsClusterExist() (bool, []string, error) {
+	ids := make([]string, 0)
+
+	if p.c == nil {
+		if err := p.generateClientSDK(); err != nil {
+			return false, ids, err
+		}
+	}
+	response, err := p.describeInstances()
+	if err != nil {
+		return false, ids, nil
+	}
+	if len(response.Response.InstanceSet) > 0 {
+		for _, resource := range response.Response.InstanceSet {
+			ids = append(ids, *resource.InstanceId)
+		}
+	}
+	return true, ids, err
+}
+
+func (p *Tencent) GenerateMasterExtraArgs(cluster *types.Cluster, master types.Node) string {
+	if option, ok := cluster.Options.(tencent.Options); ok {
+		if cluster.CloudControllerManager {
+			extraArgs := fmt.Sprintf(" --kubelet-arg=provider-id=%s.%s,--cloud-provider=external,--node-status-update-frequency=30s --node-name=%s",
+				option.Region, master.InstanceID, master.InternalIPAddress[0])
```

Comment:
add `--cloud-provider=external,--node-status-update-frequency=30s` for `kubelet`
change nodename to node private ip